client/me/purchases/manage-purchase/purchase-meta.jsx

```diff
@@ -284,52 +283,53 @@ function PurchaseMetaPrice( { purchase } ) {
 		return translate( 'Free with Plan' );
 	}
 
-	if ( purchase.billPeriodLabel ) {
-		switch ( purchase.billPeriodLabel ) {
-			case 'per year':
-				period = translate( 'year' );
```

Comment:
When the previous statements here were reassigning `period`, the one that took priority was the _last_ assignment, but the`getPeriod()` function returns early instead, meaning that the one that will take priority will be the _first_ assignment. So in order to preserve the same logic, you'll need to reverse the order of the conditions inside the function.

To explain better, let's say (this is an unlikely example but it could happen) that `purchase.billPeriodLabel` is `'per month'` and `plan.term` is `TERM_BIENNIALLY`. Before your changes, first `period` would be set to `'year'` on line 290, then it would be set to `'two years'` on line 307. After your changes, line 290 would return `'year'` instead, totally ignoring `plan.term`.

Reply:
That makes sense. Let me have this done!.